spellcheck/src/speller.c

```diff
@@ -203,13 +203,28 @@ gint sc_speller_process_line(GeanyDocument *doc, gint line_number, const gchar *
 {
 	gint pos_start, pos_end;
 	gint wstart, wend;
-	GString *str;
 	gint suggestions_found = 0;
+	gint wordchars_len;
+	gchar *wordchars_orig;
+	GString *str;
 
 	g_return_val_if_fail(sc_speller_dict != NULL, 0);
 	g_return_val_if_fail(doc != NULL, 0);
 	g_return_val_if_fail(line != NULL, 0);
 
+	/* add ' (single quote) temporarily to wordchars
+	 * to be able to check for "doesn't", "isn't" and similar */
+	wordchars_len = scintilla_send_message(doc->editor->sci, SCI_GETWORDCHARS, 0, 0);
+	wordchars_orig = g_malloc0(wordchars_len + 1);
+	scintilla_send_message(doc->editor->sci, SCI_GETWORDCHARS, 0, (sptr_t)wordchars_orig);
+	if (! strchr(wordchars_orig, '\''))
+	{
+		GString *wordchars_new = g_string_new(wordchars_orig);
```

Comment:
depending on whether it's a hot spot, you could also simply add the `'` to the orig string and truncate the last byte afterward (instead of copying the string)